src/mon/MgrMonitor.cc

```diff
@@ -107,6 +108,16 @@ void MgrMonitor::create_pending()
 {
   pending_map = map;
   pending_map.epoch++;
+
+  if (map.get_epoch() == 1 &&
+      command_descs.empty() &&
+      pending_command_descs.empty()) {
+    // we've been through the initial map and we haven't populated the
+    // command_descs vector. This likely means we came from kraken, where
+    // we wouldn't populate the vector, nor would we write it to disk, on
+    // create_initial().
+    create_initial();
```

Comment:
yes, we should have this in luminous. but i am not sure if we need it in master (i.e. mimic) if our upgrade path is something like: jewel/kraken => luminous (not optional) => mimic.

Reply:
@liewegas can you please confirm what kefu mentioned about luminous being a required step in the upgrade path from kraken to mimic? if this is confirmed, I'll be happy to have this patch targeting solely luminous.